src/PAModel/SourceTransforms/AppTestTransform.cs

```diff
@@ -133,57 +143,86 @@ public void BeforeWrite(BlockNode control)
             foreach (var child in control.Children)
             {
                 var propName = child.Name.Identifier;
+                
                 if (child.Name.Kind.TypeName != _testStepTemplateName)
                 {
                     _errors.ValidationError($"Only controls of type {_testStepTemplateName} are valid children of a TestCase");
                     throw new DocumentException();
                 }
+                
                 if (child.Properties.Count > 3)
                 {
                     _errors.ValidationError($"Test Step {propName} has unexpected properties");
                 }
+
                 var descriptionProp = child.Properties.FirstOrDefault(prop => prop.Identifier == "Description");
+                
                 if (descriptionProp == null)
                 {
                     _errors.ValidationError($"Test Step {propName} is missing a Description property");
                     throw new DocumentException();
                 }
+                
                 var valueProp = child.Properties.FirstOrDefault(prop => prop.Identifier == "Value");
+                
                 if (valueProp == null)
                 {
                     _errors.ValidationError($"Test Step {propName} is missing a Value property");
                     throw new DocumentException();
                 }
+                
                 var screenProp = child.Properties.FirstOrDefault(prop => prop.Identifier == "Screen");
 
                 string screenId = null;
+                
                 // Lookup screenID by Name
                 if (screenProp != null && !_screenIdToScreenName.ToDictionary(kvp => kvp.Value, kvp => kvp.Key).TryGetValue(screenProp.Expression.Expression, out screenId))
                 {
                     _errors.ValidationError($"Test Step {propName} references screen {screenProp.Expression.Expression} that is not present in the app");
                     throw new DocumentException();
                 }
-
-                testStepsMetadata.Add(new TestStepsMetadataJson()
+                
+                // If TestStepsMetadata exists, add
+                if (_entropy.DoesTestStepsMetadataExist == null)
```

Comment:
Since you're making this same check in 2 places, just store it in a local variable and use the `??` for null coalescing. So something like 
```cs
bool doesTestStepsMetadataExist = _entropy.DoesTestStepsMetadataExist ?? true;
```